language-plutus-core/src/Language/PlutusCore/MkPlc.hs

```diff
@@ -28,14 +28,14 @@ mkIterApp
     :: Term tyname name ()
     -> [Term tyname name ()]
     -> Term tyname name ()
-mkIterApp fun args = foldl' (Apply ()) fun args
+mkIterApp = foldl' (Apply ())
```

Comment:
I find this significantly less clear. If you really want to do this then please haddock all the arguments so it's clear what they are.